pkg/controller/apimanagerrestore/apimanagerrestore_logic_reconciler.go

```diff
@@ -321,6 +330,21 @@ func (r *APIManagerRestoreLogicReconciler) apiManagerFromSharedBackupSecret() (*
 }
 
 func (r *APIManagerRestoreLogicReconciler) reconcileRestoreAPIManager() (reconcile.Result, error) {
+	apiManagerToRestoreName := r.cr.Status.APIManagerToRestoreRef.Name
```

Comment:
At this point, if there are no programming errors APIManagerToRestoreRef should never be nil and Name should be a non-empty string. Thus, no checks related to that are performed each time the attribute is referenced in steps that are after the step that should set this status value

Reply:
The reason I haven't add it as a comment is due to it appears in several places and it's kind of a "way of proceeding".
Adding it on all occurrences each time we reference the attribute can be kind of verbose. What do you want to do?

Reply:
Not straightforward/obvious comments are always welcome.

Reply:
I added it on the first occurrencly only. There's no point on keep repeating the same comment each time that attribute is referenced